resources/views/livewire/administrative-tools/migrate-achievements.blade.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+use App\Models\Achievement;
+use App\Models\Game;
+use App\Platform\Actions\MigrateAchievementIdsToDifferentGameId;
+use Filament\Forms;
+use Filament\Forms\Concerns\InteractsWithForms;
+use Filament\Forms\Contracts\HasForms;
+use Filament\Forms\Form;
+use Filament\Notifications\Notification;
+use Illuminate\Support\Facades\Auth;
+use Livewire\Volt\Component;
+
+new class extends Component implements HasForms {
+    use InteractsWithForms;
+
+    public array $achievementIds = [];
+    public ?int $gameId = null;
+
+    public function submit(): void
+    {
+        // Validate.
+        $this->form->getState();
+
+        /** @var User $user */
+        $user = Auth::user();
+
+        (new MigrateAchievementIdsToDifferentGameId())->execute(
+            $this->achievementIds,
+            $this->gameId,
+            $user,
+        );
+
+        Notification::make()
+            ->success()
+            ->title('Success')
+            ->body('Migrated!')
+            ->send();
+
+        // Reset the form.
+        $this->form->fill();
+        $this->achievementIds = [];
+        $this->gameId = null;
+    }
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Forms\Components\Select::make('achievements')
+                    ->label('Achievements')
+                    ->placeholder('Select achievements')
+                    ->multiple()
+                    ->searchable()
+                    ->getSearchResultsUsing(function (string $search): array {
+                        return Achievement::with('game')
+                            ->where('Title', 'like', "%{$search}%")
+                            ->orWhere('ID', 'like', "%{$search}%")
+                            ->limit(50)
+                            ->get()
+                            ->mapWithKeys(function ($achievement) {
+                                return [$achievement->id => "ID: {$achievement->id} - Title: {$achievement->title} - Game: {$achievement->game->title}"];
+                            })
+                            ->toArray();
+                    })
```

Comment:
I don't know that this feature gets used all that often, but I imagine that when it is used, people just paste in a CSV of achievements to move. CSVs don't seem to work with the new component. You have to enter each ID individually, and select the matching result.

Reply:
Updated in latest to use a CSV input field.